lib/Conversion/MooreToCore/MooreToCore.cpp

```diff
@@ -1174,6 +1174,111 @@ struct ShrOpConversion : public OpConversionPattern<ShrOp> {
   }
 };
 
+struct PowUOpConversion : public OpConversionPattern<PowUOp> {
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(PowUOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    Type resultType = typeConverter->convertType(op.getResult().getType());
+
+    Location loc = op.getLoc();
+    auto intType = cast<IntType>(op.getRhs().getType());
+    if (auto baseOp = op.getLhs().getDefiningOp<ConstantOp>()) {
+      if (baseOp.getValue() == 2) {
+        Value constOne = rewriter.create<ConstantOp>(loc, intType, 1);
+        rewriter.replaceOpWithNewOp<ShlOp>(op, constOne, op.getRhs());
+        return success();
+      }
+    }
```

Comment:
This would be fantastic to have as a canonicalization in the Moore dialect! Could you move this into the canonicalizer of `moore::PowUOp`? (You might have to add a `let hasCanonicalizeMethod = 1` to the `PowUOp` definition in the `MooreOps.td` file.)